ts-eks/test/pulumi/index.ts

```diff
@@ -15,14 +15,32 @@ const eksPolicy: StackValidationPolicy = {
         const cluster = eksClusters[0].asType(aws.eks.Cluster)
 
         const nodeGroups = args.resources.filter(r => r.isType(aws.eks.NodeGroup));
-        if (nodeGroups.length !== 1) {
-            reportViolation(`Node groups: expected 1 but found ${nodeGroups.length}`);
+        if (nodeGroups.length !== 2) {
+            reportViolation(`Expected three Node Groups but found ${nodeGroups.length}`);
             return;
         }
 
-        const group = nodeGroups[0].asType(aws.eks.NodeGroup)!;
-        if (group.clusterName != cluster?.name) {
-            reportViolation(`Expected cluster name for node group to be ${cluster?.name}, but found ${group.clusterName}`)
+        let c4LargeSubnetIds: any[] = [];
+        let t3LargeSubnetIds: any[] = [];
+        for (const i in nodeGroups) {
+            const group = nodeGroups[i].asType(aws.eks.NodeGroup)!;
+
+            if (group.clusterName != cluster?.name) {
+                reportViolation(`Expected cluster name for node group to be ${cluster?.name}, but found ${group.clusterName}`)
+            }
+
+            if (group.nodeGroupName.includes("c4-large")) {
+                group.diskSize !== 100 ?  reportViolation(`Expected disk size to be '100' fo 'c4.large' node group, but found ${group.diskSize}`) : null
```

Comment:
the default disk size is 20. is the auto scaling going to immediately trigger and set it to 100?

Reply:
similar to the answer above, we figure even though the c4.large specification only requires 20, the other exec unit with no instance type specification requires 100 so we just create 1 node group where its 100. The thought is, that its cheaper to have 2 instances with more disk space than double the amount of instances. 

We also dont have cluster autoscaling yet so no real autoscaling